smb/src/resource.rs

```diff
@@ -269,6 +276,13 @@ impl ResourceHandle {
         self.share_type
     }
 
+    fn file_id(&self) -> crate::Result<FileId> {
+        if !self.open.load(std::sync::atomic::Ordering::SeqCst) {
+            return Err(Error::InvalidState("Resource is closed".into()));
+        }
+        Ok(self._file_id)
+    }
+
```

Comment:
This doesn't provide the guarantees you likely think it does. There is nothing stopping open from being set to false immediately after loading it. That said, I don't see that this race exists in your code because of the reasons I cited above.

All atomic operations in this file can be `Ordering::Relaxed` (as I said above, I don't think you even need an atomic).